crates/red_knot_python_semantic/src/types.rs

```diff
@@ -678,6 +700,13 @@ impl<'db> Type<'db> {
                 | Type::ClassLiteral(..)),
             ) => left != right,
 
+            (Type::Type(type_class), Type::ClassLiteral(class_literal))
+            | (Type::ClassLiteral(class_literal), Type::Type(type_class)) => {
+                !class_literal.class.is_subclass_of(db, type_class.class)
+            }
+            (Type::Type(_), Type::Type(_)) => false,
+            (Type::Type(_), Type::Instance(_)) | (Type::Instance(_), Type::Type(_)) => false,
+            (Type::Type(_), _) | (_, Type::Type(_)) => false,
```

Comment:
I think we are missing a case to establish that `Type::Type(_)` is disjoint from all literal types that are not `ClassLiteral`?